File: apps/ErlangCowboy/src/ws_echo_handler.erl
-module(ws_echo_handler).
-behaviour(cowboy_websocket).

-export([init/2, websocket_handle/2, websocket_info/2]).

init(Req, State) ->
    {cowboy_websocket, Req, State}.

websocket_handle({text, Msg}, State) ->
    {reply, {text, Msg}, State};
websocket_handle(_, State) ->
    {ok, State}.

websocket_info(_, State) ->
    {ok, State}.

File: apps/ElixirPhoenix/lib/elixir_phoenix_api/ws_echo_handler.ex
defmodule ElixirPhoenixApi.WsEchoHandler do
  @behaviour :cowboy_websocket

  def init(req, state) do
    {:cowboy_websocket, req, state, %{idle_timeout: 60_000}}
  end

  def websocket_handle({:text, msg}, state) do
    {:reply, {:text, msg}, state}
  end

  def websocket_handle(_, state) do
    {:ok, state}
  end

  def websocket_info(_, state) do
    {:ok, state}
  end
end
